### ISSUE TITLE
Show token capabilities by policy

### ISSUE THREAD

**Issue by mdgreenfield** [hashicorp/vault#7235]
**Describe the solution you'd like**
As a Vault admin, I'd like to be able to quickly map a request path for given token to the capabilities offered by each policy to quickly isolate _what policy_ is allowing/denying access to a given path.

The current `sys/capabilities*` endpoints only display an aggregate of all capabilities a token has but would like to see the capabilities broken down by policy.

**Describe alternatives you've considered**
If admins are managing their policies via CM, an alternative here is to grep the policies for a given path (or subpath). However, searching policies in CM can get complex if templated policies are used making it harder for admins to track down a polity allowing/denying access.

If admins are not managing their policies in CM, they'd have to read/download those policies first.

**Additional context**
Because the `default` policy does not allow users to read/list policies it probably makes sense to implement this functionality in a new endpoint so that admins can restrict access to this endpoint appropriately. This way, users do not gain unwanted insight into the policies they have associated to their token. 

If this endpoint were `sys/provenance` it would probably make sense to also expose a `sys/provenance-accessor`, `sys/provenance-self` endpoint as well for admins to allow access as appropriate based on their security needs.

**Comment by aphorise** [comment]
Hey @mdgreenfield - I know it's been a long time - but [I believe that **`-output-policy`**](https://www.vaultproject.io/docs/commands#print-policy-requirements) that was recently introduced in 1.11.0 does exactly this. Can you kindly confirm? - thanks.

**Comment by mdgreenfield** [comment]
Hi @aphorise, I hadn't seen the work for the `-output-policy` flag before. It looks useful for translating CLI commands into policy statements but it looks to be a different use case.

Re-reading my original issue, I was hoping to have support built into Vault that would basically allow an admin to supply a token accessor and a Vault path (and probably method) and have Vault return the policy by name that would accept/reject the request (or no policy if there is no matching policy). This is particularly useful for Vault deployments where policy wildcards are used (both `*` and `+`).

**Comment by aphorise** [comment]
@mdgreenfield - I think I get what you're getting at. So for any give path watch are the matching policies anywhere that could match? A sort of traverse matching routes and the related policies?

**Comment by mdgreenfield** [comment]
Correct. This would give an admin/user the ability to quickly identify what policy is causing a request to be accepted/rejected.

**Comment by aphorise** [comment]
@mdgreenfield on a related side note - this is actually one reason why I'd advocate for no-default-policy and opting instead for explicit merged single policy per group; this itself reduces the set of policies to evaluate without any scope consideration between policies or default vs overlapping or conflict elsewhere.

There is also the case of generic paths (from default policy) or root exclusions too which can be a major exception particularly if you are not going by no-root rule or limiting to a single token only.

Thinking aloud - something may already be possible to achieve this [using Sentinel](https://learn.hashicorp.com/tutorials/vault/sentinel-http-import?in=vault/enterprise) although it may be a bit of a hack 😆
